### ISSUE TITLE
[Major Feature] Training on surface relative transforms

### ISSUE THREAD

**Issue by ahundt** [jhu-lcsr/costar_plan#296]
Train on surface relative transforms 
-------

<img width="1511" alt="2017-12-16 surface relative transforms correct" src="https://user-images.githubusercontent.com/55744/34134058-5846b59e-e426-11e7-92d6-699883199255.png">

[Figure 6 in the Amazon Proposal pdf document](https://www.overleaf.com/read/kvhyvkynjztz) illustrates surface relative transforms to gripper positions, when this task is complete we will be able to train using the grasp dataset's raw depth image and robot pose information.

Steps to complete this task:

- [x] #293 visualize point cloud in v-rep, include a function to add points and to clear the points from a v-rep point cloud
- [x] #292 create and configure a function to filter raw depth images so failed pixels (depth 0) are reduced or eliminated
- [x] create a function [def depth_image_pixel_to_cloud_point(depth_image, camera_intrinsics_matrix, pixel_coordinate, augmentation_rectangle=None)](https://github.com/cpaxton/costar_plan/blob/master/costar_google_brainrobotdata/grasp_geometry.py#L172) that can calculate xyz positions in the image frame when given a depth image and image intrinsic parameters
- [x] #291 Image Augmentation - Add support for random cropping of both an rgb and depth image, and integrate it into the training script.
- [x] #294 Surface relative transform calculation + visualization - calculate the transforms for the grasping dataset. see [grasp-viz releases](https://github.com/ahundt/costar_plan/releases) for details.
- [x] https://github.com/cpaxton/costar_plan/issues/360 Image resizing
- [x] https://github.com/cpaxton/costar_plan/issues/366 integrate pixel-wise loss into grasp_train.py
- [x] https://github.com/cpaxton/costar_plan/issues/365 fix resnet model nan loss
- [x] #295 Add pixel-wise options to grasp model, Integrate Spatial Softmax 
- [x] #297  100 Epoch Training Run - Spatial Grasp Prediction

The [Latest version of the V-REP scene file](https://github.com/ahundt/costar_plan/releases/tag/grasp_viz_0.1) necessary for visualization is available in the fork of @ahundt. Code is currently maintained in the [grasp_train_quaternion](https://github.com/cpaxton/costar_plan/tree/grasp_train_quaternion) branch.
